dds_web/security/project_user_keys.py

```diff
@@ -41,43 +71,47 @@ def __decrypt_with_rsa(ciphertext, private_key):
 
 
 def __encrypt_project_private_key(owner, project_private_key):
-    public_key = serialization.load_der_public_key(owner.public_key)
-    if isinstance(public_key, asymmetric.rsa.RSAPublicKey):
-        return __encrypt_with_rsa(project_private_key, public_key)
-    # TODO: Change exception type
-    exception = Exception("Public key cannot be loaded for encrypting the project private key!")
-    flask.current_app.logger.exception(exception)
-    raise exception
-
-
-def __decrypt_project_private_key(user, encrypted_project_private_key):
-    user_private_key = serialization.load_der_private_key(
-        __decrypt_user_private_key(user), password=None
-    )
-    if isinstance(user_private_key, asymmetric.rsa.RSAPrivateKey):
-        return __decrypt_with_rsa(encrypted_project_private_key, user_private_key)
-    exception = Exception("User private key cannot be loaded!")
-    flask.current_app.logger.exception(exception)
-    raise exception
+    if not owner.public_key:
+        raise KeySetupError(message="User keys are not properly setup!")
+
+    try:
+        owner_public_key = serialization.load_der_public_key(owner.public_key)
+        if isinstance(owner_public_key, asymmetric.rsa.RSAPublicKey):
```

Comment:
Should we have an else case here that raises an exception?

Reply:
I think the if statement is the creation of my cautiousness. I think if it is not a public key, `load_der_public_key` will raise an exception, so just to be on the safe side. I don't think we need an else here.